core/src/main/java/org/apache/accumulo/core/util/HostAndPort.java

```diff
@@ -281,4 +282,16 @@ public int getPortOrDefault(int defaultPort) {
     return hasPort() ? port : defaultPort;
   }
 
+  /**
+   * HostAndPort must implement compareTo. As this is a seldom used utility, compareTo simply treats
+   * HostAndPort values as Strings.
+   */
+  @Override
+  public int compareTo(HostAndPort other) {
+    return Comparator
+        .nullsFirst(
+            Comparator.comparing(HostAndPort::getHost).thenComparingInt(h -> h.getPortOrDefault(0)))
```

Comment:
Does this still work if host is null? From the constructor, it doesn't look like we require non-null hosts. This might fail to compare if the host is null.